### ISSUE TITLE
Hidden toctree cause validation problems with EPUB

### ISSUE THREAD

**Issue by shimizukawa** [sphinx-doc/sphinx#1323]
When you hide toctree entries somewhere in your project (not in index.rst) you get an validation error from epubcheck due to an empty ul-enviroment.

```
#!bash

ERROR: _build/epub/Sphinx.epub/index.html(79,6): element "ul" incomplete; missing required element "li"
```

The hidden toctree entry is in the file with the headline "Einsatzgebiete".

Attached is the file index.html

---
- Bitbucket: https://bitbucket.org/birkenfeld/sphinx/issue/1323
- Originally reported by: Jan Ulrich Hasecke
- Originally created at: 2013-12-14T20:30:27.437

**Comment by shimizukawa** [comment]
_From Roland Meister on 2014-01-10 21:00:12+00:00_

Yes, also kindlegen complains about the empty element.
I'll try to add a transform that removes empty elements from
the docutils tree for the html builder.

**Comment by shimizukawa** [comment]
_From Georg Brandl on 2014-01-19 17:29:33+00:00_

Closes #1323: Fix emitting empty `<ul>` tags in the HTML writer, which is not valid HTML.

→ <<cset 7a74fe81b18f18d38cf3820fe3154a3279219ffc>>
